app/src/play/java/de/danoeh/antennapod/preferences/PreferenceControllerFlavorHelper.java

```diff
@@ -26,7 +32,16 @@ public static void setupFlavoredUI(PlaybackPreferencesFragment ui) {
                     return false;
                 }
             }
+            displayRestartRequiredDialog(ui.requireContext());
             return true;
         });
     }
+
+    private static void displayRestartRequiredDialog(@NonNull Context context) {
+        AlertDialog.Builder dialog = new AlertDialog.Builder(context);
+        dialog.setTitle(android.R.string.dialog_alert_title);
+        dialog.setMessage(R.string.pref_restart_required);
+        dialog.setPositiveButton(android.R.string.ok, null);
+        dialog.show();
```

Comment:
I think the dialog should not be cancelled when touching the outside. Back arrow and ok button are fine but I think it is too important to be cancelled accidentally.

Reply:
make sense - should we do the same for the similar prompt in image cache size settings?

Reply:
I don't think it is necessary for the cache size. Users don't see the image cache size. If they miss to restart, they won't notice. That's different with casting: If they accidentally skip the dialog, they will go back and wonder why it does not work.

Reply:
ok.